types/fs-base-ui/html_ui/JS/SimPlane.d.ts

```diff
@@ -0,0 +1,421 @@
+         */
+        function getV1Airspeed(): Knots | -1  | null;
+
+        /**
+         * Gets the VR speed up during and before takeoff, -1 after.
+         */
+        function getVRAirspeed(): Knots | -1  | null;
+
+        /**
+         * Gets the VR speed up during and before takeoff, -1 after.
+         */
+        function getV2Airspeed(): Knots | -1 | null;
+        function getREFAirspeed(): Knots | null;
+        function getVXAirspeed(): Knots | null;
+        function getFMCGreenDotSpeed(): number | null;
+        function getFMCApprGreenDotSpeed(): number | null;
+        function getGreenDotSpeed(): Knots | null;
+        function getCrossoverSpeedFactor(cas, mach: Mach): number | null;
+        function getFlapsLimitSpeed(aicraft: any, flapIndex: number): Knots | null;
+        /**
+         * @summary
+         * Seems to implement caching behaviour, can be overridden with forceSimVarCall.
+         */
+        function getFlapsHandleIndex(forceSimVarCall?: boolean): number | null;
+        /**
+         * @summary
+         * Seems to implement caching behaviour, can be overridden with forceSimVarCall.
+         */
+        function getFlapsExtendSpeed(aircraft: any, forceSimVarCall?: boolean): Knots | null;
+        /**
+         * @summary
+         * Seems to implement caching behaviour, can be overridden with forceSimVarCall.
+         */
+        function getNextFlapsExtendSpeed(aircraft: any, forceSimVarCall?: boolean): Knots | null;
+        function getMaxSpeed(aircraft: any): Knots;
+        function getLowestSelectableSpeed(): Knots | null;
+        function getStallProtectionMinSpeed(): Knots | null;
+        function getStallProtectionMaxSpeed(): Knots | null;
+        function getStallSpeed(): Knots | null;
+        function getStallSpeedPredicted(flapIndex: number): Knots | null;
+        function getWindDirection(): Degrees | null;
+        function getWindStrength(): Knots | null;
+        function getAutoPilotActive(apIndex: number): boolean | null;
+        function getAutoPilotAirspeedManaged(): boolean;
+        function getAutoPilotAirspeedSelected(): boolean;
+        function getAutoPilotAirspeedHoldActive(isManaged?: boolean): boolean | null;
+        function getAutoPilotAirspeedHoldValue(): Knots | null;
+        function getAutoPilotSelectedAirspeedHoldValue(): Knots | null;
+        function getAutoPilotManagedAirspeedHoldValue(): Knots | null;
+        function getAutoPilotMachModeActive(): boolean | null;
+        function getAutoPilotMachHoldValue(): number | null;
+        function getAutoPilotSelectedMachHoldValue(): number | null;
+        function getAutoPilotManagedMachHoldValue(): number | null;
+        function getAutoPilotHeadingManaged(): boolean;
+        function getAutoPilotHeadingSelected(): boolean;
+        function getAutoPilotHeadingLockActive(): boolean | null;
+        function getAutoPilotHeadingLockValue(radians?: boolean): Radians | Degrees | null;
```

Comment:
Maybe this can be restricted with boolean literal types:
```ts
        function getAutoPilotHeadingLockValue(radians?: true): Radians | null;
        function getAutoPilotHeadingLockValue(radians?: false): Degrees | null;
```

Reply:
I think it's a valuable idea and thus wrote:

```typescript
function getAutoPilotHeadingLockValue(): Radians | null;
function getAutoPilotHeadingLockValue(radians: true): Radians | null;
function getAutoPilotHeadingLockValue(radians: false): Degrees | null;
```

However, sadly only the first two are correctly "identified" by WebStorm. When I pass false it will still point to the second (true) overload. Thus I've decided to stick to the current version for now.